frontend/readhistory.lua

```diff
@@ -268,20 +268,24 @@ end
 --- Adds new item (last opened document) to the top of the history list.
 function ReadHistory:addItem(file, ts)
     if file ~= nil and lfs.attributes(file, "mode") == "file" then
-        local index = self:getIndexByFile(realpath(file))
-        if index then
-            table.remove(self.hist, index)
-        end
-        local now = ts or os.time()
-        table.insert(self.hist, 1, buildEntry(now, file))
         -- util.execute("/bin/touch", "-a", file)
         -- This emulates `touch -a` in LuaFileSystem's API, since it may be absent (Android)
         -- or provided by busybox, which doesn't support the `-a` flag.
```

Comment:
Unrelated to your endeavors, but feel free to remove that comment, because using lfs was always the sanest approach vs. spawning `touch`, regardless of the platform ;).